### ISSUE TITLE
`JSON.stringify` is converting `Float` (`number`) values to strings.

### ISSUE THREAD

**Issue by ifightcrime** [prisma/prisma#12471]
### Bug description

I'm using prisma with Typescript in a next.js application and JSON.strinfigy converts fields that are of type `number` (`Float` in the schema) to strings.

I did a little debugging and it looks like the `Float` fields are actually `typeof 'object'`, not `number`, as I would have expected and what the generated `@prisma/client` types are telling me. The only thing I can think of is that Prisma is wrapping these values in something funky in the background?

For now my workaround is to do something like this:

```typescript
export const toRaw = (user: User) => ({
  id: user.id,
  ...
  name: user.name,
  latitude: Number(user.latitude),
  longitude: Number(user.longitude),
});
```

### How to reproduce

When I fetch and try to return json, `latitude` and `longitude` are being converted to strings:

```typescript
const user = await prisma.user.findUnique({
  where: {
    id: req.user.id,
  },
});

JSON.stringify(user);
```

The output of which is:

```json
{
  "name": "Super Dude",
  "longitude": "-118.235322", // <- STRING, not a number
  "latitude": "34.045148",  // <- STRING, not a number
}
```

### Expected behavior

I'd expect the output to be:

```json
{
  "name": "Super Dude",
  "longitude": -118.235322, // <- NUMBER, not a string
  "latitude": 34.045148,  // <- NUMBER, not a string
}
```

### Prisma information

My schema for a `User` model is as follows:

```
// Schema
model User {
  id            String      @id @default(cuid())
  ...
  name          String?
  latitude      Float?
  longitude     Float?
}

// Generated type in @prisma/client
/**
 * Model User
 * 
 */
export type User = {
  id: string
  ...
  name: string | null
  latitude: number | null
  longitude: number | null
}
```

### Environment & setup

Next.js with Typescript.

### Prisma Version

```
prisma                  : 3.10.0
@prisma/client          : 3.10.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 73e60b76d394f8d37d8ebd1f8918c79029f0db86
Studio                  : 0.458.0
```

**Comment by pantharshit00** [comment]
I am unable to reproduce this. I ran the query that you have posted and ran `JSON.stringify` to the result and output was correct:
<img width="882" alt="image" src="https://user-images.githubusercontent.com/22195362/159455422-58c33640-4c39-42b7-9f4d-aa396ceae801.png">


I used Postgres as database? Can you please try again with latest version and possibly report which database are you using?

**Comment by pantharshit00** [comment]
Script used:
```ts
import { PrismaClient } from "@prisma/client";

async function main() {
  const prisma = new PrismaClient();

  const user = await prisma.user.findUnique({
    where: {
      id: "cl11yqszv00072311gdwx6vo7",
    },
  });

  console.log(JSON.stringify(user));
  await prisma.$disconnect();
}

main();
```

**Comment by janpio** [comment]
Ping @ifightcrime. Can you help us out here a bit? We have no idea how to reproduce this to understand and potentially fix.

**Comment by ifightcrime** [comment]
@janpio hey yes sorry I missed the first message! At the time I was a number of versions back too so who knows. I'm about to head out for the weekend though, but I will get back to you likely early next week.

**Comment by janpio** [comment]
Did you have any time to look at this @ifightcrime maybe?

**Comment by janpio** [comment]
Can you help us out here @ifightcrime? With the current information we unfortunately will need to close the issue as it is not enough to understand what is going on and reproduce it. Thanks!

**Comment by qcom** [comment]
Might be a duplicate of #6049?

@janpio I encountered similar behavior today and it really turned me around for a minute before I realized that prisma was returning decimal.js objects instead of native JavaScript numbers.

I use SQL Server, for reference. When copying the example from this issue, everything was working as correctly (i.e. `JSON.stringify()` correctly processing latitude and longitude as numbers). Only once I changed those fields from `float` to `decimal` did I notice the behavior return. I'm using `decimal` data types in my project, so that makes sense as to why I also noticed it originally.

Ultimately I understand why prisma elects to wrap numbers, as I too use decimal.js for important calculations in JavaScript to avoid unexpected floating point issues. That said it definitely surprised me when I was inspecting my objects in the debugger and pre-stringify they were evaluating as numbers and post-stringify they were strings. It took me running `typeof` to notice it was an object, then check its constructor which made it clear it was decimal.js.

I'll definitely be implementing some kind of workaround in my project. Possibly [this solution](https://github.com/prisma/prisma/issues/6049#issuecomment-1132894878) but I'm hesitant to overwrite the prisma prototype as such.

See below for my `schema.prisma` and simple script. Happy to share the repo as well if it would help.

### index.ts
```ts
import { PrismaClient } from "@prisma/client";

async function main() {
  const prisma = new PrismaClient();
  const name = 'foo';
  const latitude = 34.045148;
  const longitude = 118.235322;

  const userResult = await prisma.user.upsert({
    where: { name },
    update: { latitude, longitude },
    create: { name, latitude, longitude },
  });

  const user = await prisma.user.findFirst({
    where: { name },
  });

  if (user) {
    const { latitude, longitude } = JSON.parse(JSON.stringify(user));
    console.log(JSON.stringify(user)); // {"id":"cla369oo40000mfknwyso4l46","name":"foo","latitude":"34.045148","longitude":"118.235322"}
    console.log(typeof latitude); // string
    console.log(typeof longitude); // string
  }

  await prisma.$disconnect();
}

main();
```

### schema.prisma
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model User {
  id            String   @id @default(cuid())
  name          String   @unique
  latitude      Decimal? @db.Decimal(15, 6)
  longitude     Decimal? @db.Decimal(15, 6)
}
```

### Environment
**OS:** macOS Monterey 12.5.1
**Database:** SQL Server 2014 (not supported by prisma... 🤫)
**Node.js version:** v16.13.1

### `prisma -v`
```
Environment variables loaded from .env
prisma                  : 4.5.0
@prisma/client          : 4.5.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 0362da9eebca54d94c8ef5edd3b2e90af99ba452 (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.5.0-43.0362da9eebca54d94c8ef5edd3b2e90af99ba452
Default Engines Hash    : 0362da9eebca54d94c8ef5edd3b2e90af99ba452
Studio                  : 0.476.0
```

**Comment by tomashaddad** [comment]
I understand the rationale behind wrapping numbers in Decimal.js objects, but their default treatment as strings is annoying. I ran into the same problem as what @qcom describes, except with Postgres and `numeric` types (an alias for `decimal`). What exactly is the best way to deal with this? Am I meant to cast these values with `Number()` everywhere in my front end? Reluctantly Decimal.js as another dependency?

**Comment by Jolg42** [comment]
@ifightcrime Closing as we didn't hear back, we're happy to reopen it if this issue still exists in a recent version of Prisma.
